modules/acquia_cms_headless/tests/src/Functional/HeadlessSubrequestsTest.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Drupal\Tests\acquia_cms_headless\Functional;
+
+use Acquia\DrupalEnvironmentDetector\AcquiaDrupalEnvironmentDetector;
+use Drupal\Tests\BrowserTestBase;
+
+/**
+ * Base class for the Headless Content administrator browser tests.
+ */
+class HeadlessSubrequestsTest extends BrowserTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaultTheme = 'stark';
+
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  protected static $modules = [
+    'acquia_cms_headless',
+    'node',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    // @todo Remove this check when Acquia Cloud IDEs support running functional
+    // JavaScript tests.
+    if (AcquiaDrupalEnvironmentDetector::isAhIdeEnv()) {
+      $this->markTestSkipped('This test cannot run in an Acquia Cloud IDE.');
+    }
+    parent::setUp();
+    $account = $this->drupalCreateUser();
+    $account->addRole('administrator');
+    $account->save();
+    $this->drupalLogin($account);
+
+    // Set up a content type.
+    $this->drupalCreateContentType([
+      'type' => 'test',
+      'name' => 'Test',
+    ]);
+  }
+
+  /**
+   * Content admin test.
+   */
+  public function testSubrequests(): void {
+    $host = \Drupal::request()->getSchemeAndHttpHost();
+    $ch = curl_init();
```

Comment:
I don't feel too strongly about this since this has been already implemented but Drupal core ships with a HTTP Client (Guzzle) which makes executing requests a bit nicer. I'm not sure why the tests are failing in CI but moving to using the Drupal core HTTP client _might_ help with that.

Reply:
We did the same with HTTP client [here](https://github.com/acquia/acquia_cms/pull/1499/commits/f155a2761d87c4b0050812c6ff133b2f1d17752b), will revert the recent commit with cURL in other open [PR](https://github.com/acquia/acquia_cms/pull/1493).